### ISSUE TITLE
Handle empty include dirs

### ISSUE THREAD

**Issue by GerrietReents** [Caphyon/clang-power-tools#129]
Our props files contain include paths that are evaluated to an empty string. Please check this special case in ´clang-buils.ps1´ like this:

```
Function Get-ProjectAdditionalIncludes([Parameter(Mandatory=$true)][string] $vcxprojPath)
{
  $data = Select-ProjectNodes $kVcxprojXpathAdditionalIncludes
  [string[]] $tokens = ($data).InnerText -split ";"
  if (!$tokens)
  {
    return
  }

  [string] $projDir = Get-FileDirectory($vcxprojPath)
  
  foreach ($token in $tokens)
  {
    if (![string]::IsNullOrWhiteSpace($token)) {
		Write-Debug "Canonize path $token"
		[string] $includePath = Canonize-Path -base $projDir -child $token -ignoreErrors
		if (![string]::IsNullOrEmpty($includePath))
		{
		  $includePath
		}
	}
  }
}
```

**Comment by hero101111** [comment]
I've analysed this scenario and looks like you've uncovered a corner case when accessing a project setting consisting of semicolon separated values.

We already took care to eliminate empty tokens, but _before_ MsBuild property expansion, instead of _after_, as it preceded this improvement. This was a bug.

Pushed a [fix](https://github.com/Caphyon/clang-power-tools/commit/f7559ebf9c491c8ba09f35df1e1f8daaf6f4f079) that addresses all property retriavals, not just include paths.
It should fix your particular issue. 

Please give it a test and let us know if it solves your problem.

**Comment by GerrietReents** [comment]
Yes, the problem is a `;;` in ´AdditionalIncludes´, but unfortunately you fix does not help.

**Comment by hero101111** [comment]
This issue should now be fixed as part of #132.
Please let us know if the issue is still reproducible.

-----

This fix is currently live only for the github repository.
The latest Visual Studio extension (2.0) does not contain the fix. It will be included in the next release.

Until then, as a workaround, you can manually replace `clang-build.ps1` in the VS extension directory with the latest from github. To locate the extension directory search for `clang-build.ps1` in `%LOCALAPPDATA%`

Don't forget to restart Visual Studio after replacing the script file.

**Comment by GerrietReents** [comment]
Yes, the problem is fixed now. Thanks for your great work!
